src/renderer/components/Settings/index.js

```diff
@@ -0,0 +1,127 @@
+// @ts-check
+import * as React from 'react'
+
+// used for typedef
+// eslint-disable-next-line no-unused-vars
+import { defineMessages, MessageDescriptor } from 'react-intl'
+import AssignmentIcon from '@material-ui/icons/Assignment'
+import { makeStyles } from '@material-ui/core'
+import Fade from '@material-ui/core/Fade'
+import Paper from '@material-ui/core/Paper'
+
+import { SettingsMenu } from './SettingsMenu'
+import { ProjectConfig } from './ProjectConfig'
+import { SettingsProvider } from './SettingsContext'
+
+const m = defineMessages({
+  // Setting menu option: Project Configuration
+  projConfig: 'Project Configuration',
+  projConfiSubHeader: 'Categories, icons, and questions'
+})
+
+const FADE_DURATION = 700
+
+/** @typedef {{tabId:SettingTabId, icon:(string | React.ReactElement<any, string | React.JSXElementConstructor<any>>), label:MessageDescriptor, subHeader:MessageDescriptor}} SettingsTabs */
+
+// This is for strong typing of the props for the SettingsMenu component tab values. When we have more tabs, we can them to this type.
+/** @typedef {'ProjConfig'} SettingTabId */
+
+/** @type {SettingsTabs[]} */
+const tabs = /** @typedef {const} */ [
+  {
+    tabId: 'ProjConfig',
+    icon: <AssignmentIcon />,
+    label: m.projConfig,
+    subHeader: m.projConfiSubHeader
+  }
+]
+
+/**
+ * @typedef SettingsProp
+ * @prop {boolean} reset
+ * @prop {function(boolean):void} setReset
+ * @prop {boolean} fadeIn
+ * @prop {boolean} practiceModeOn
+ */
+
+/** @param {SettingsProp} props */
+export const Settings = ({ reset, fadeIn, practiceModeOn }) => {
+  // eslint-disable-next-line no-unused-vars
+  const [menuVisible, setMenuVisibility] = React.useState(true)
+
+  /** @type {SettingsTabs['tabId'] | false} */
+  const initialState = /** {const} */ (false)
+
+  const [tabValue, setTabValue] = React.useState(initialState)
+
+  const classes = useStyles()
+
+  return (
+    <SettingsProvider practiceModeOn={practiceModeOn}>
+      <Fade in={reset || fadeIn} timeout={FADE_DURATION}>
```

Comment:
seems like the only usage of `reset` and `fadeIn` props - perhaps we can merge as a single prop to make it less work for the parent components?

Reply:
Agreed, this will make sense later once we deal with the map component (the map component hides the tab bar, so having a reset available does make it easier). But ill get rid of it until we merge that in